VisualizationBase/src/nodes/InfoNode.cpp

```diff
@@ -52,18 +52,32 @@ InfoNode::InfoNode(Model::Node *target, QJsonArray enabledInfos)
 	Q_ASSERT(target);
 	for (auto item : enabledInfos)
 		setEnabled(item.toString(), true);
-	allInfoNodes.append(this);
+	initialize();
 }
 
 InfoNode::~InfoNode()
 {
-	allInfoNodes.removeAll(this);
+	allInfoNodes.remove(key_);
 	target_ = nullptr;
 }
 
+void InfoNode::initialize()
```

Comment:
Well in that case you can leave the `allInfoNodes` stuff as it was before, just a `QList` and please use the pointer directly in javascript.